### PR TITLE
FAI-126: Refactor trusty-ui to retrieve model details from execution id

### PR DESCRIPTION
**Pull request by manstis**
Depends on https://github.com/kiegroup/trusty-ai-sandbox/pull/20

See https://issues.redhat.com/browse/FAI-126

This PR:-
- Refactor to use function components
- Mock API for `/models/executionId` to return the XML for an execution
- Change `ModelDiagram` to render different viewers based on model type

TODO (before this is merged) 
- Add a spinner thingy while the REST call completes.
- Move DMN to a new component.

### CODE REVIEW

File: front-end-poc/src/ModelLookup/ModelDiagram.tsx [hunk 1]
```diff
@@ -1,21 +1,60 @@
...
+
+  useEffect(() => {
+    let didMount = true;
+    getModelDetail(id)
+      .then(response => {
+        if (didMount) {
+          const modelType: string = response.data[0].modelType;
+          const xml: string = response.data[0].xml;
+
+          switch (modelType) {
```
Comment (manstis) [inline front-end-poc/src/ModelLookup/ModelDiagram.tsx:40]: @kelvah Is there a better way to handle embedding the correct viewer? I return a _place holder_ DOM element then use `ReactDOM.render(..)` to set the content of the placeholder once the model type and XML have been retrieved from the endpoint.

**Comment by manstis** [comment]
Here's a couple of examples (executions 1000-1004 are DMN, 1005-1009 PMML):

**DMN**
Screenshot from 2020-04-22 16-05-54

**PMML**
Screenshot from 2020-04-22 16-05-03

I revised the "Version selector" to simply be a button to show the "Outcome". IDK what purpose/use that has now I've removed _versioning_. Can you advise?

Screenshot from 2020-04-22 16-06-48

File: front-end-poc/src/ModelLookup/ModelDiagram.tsx (hunk 1 above)
Comment (kelvah) [reply front-end-poc/src/ModelLookup/ModelDiagram.tsx:40]: @manstis The embed functions could return the `jsx` of each viewer, you could store it in a `viewer` variable inside the switch, and then use it in the return function of `ModelDiagram`, like `{viewer}`.
You might need to manage the state of the `viewer` with `useState`, given that you are manipulating it inside 'useEffect`.

**Comment by kelvah** [comment]
> I revised the "Version selector" to simply be a button to show the "Outcome". IDK what purpose/use that has now I've removed _versioning_. Can you advise?

It has none. It was meant to display or simulate the decision for the selected model version. It should be hidden along the versioning at this point.

File: front-end-poc/src/ModelLookup/ModelDiagram.tsx (hunk 1 above)
Comment (manstis) [reply front-end-poc/src/ModelLookup/ModelDiagram.tsx:36]: Great. Done.. much nicer. I also set the initial state to PatternFly's `<Spinner>`.

**Comment by manstis** [comment]
@kelvah Thank-you for your feedback.

I've added a new commit removing the `ModelOutcomeDialog` and related toolbar component (that held the button to display the dialog). I also took this opportunity to replace use of `id` (in `IExecutionRouteParams`) to `executionId` as that better reflects what it is. This led to a couple of other changes in some of the other components and `json-server` configuration. Everything seems to still be working fine!

**Comment by manstis** [comment]
Hi @kelvah I've moved the PMML _viewer_ as agreed.. everything works fine now :-)

File: front-end-poc/src/ModelLookup/pmml/LinearRegressionViewAdaptor.tsx [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+import { Line, LinearRegressionView, Range } from './LinearRegressionView';
+
+type Props = {
+  dictionary: PMML.DataDictionaryType
+  model: PMML.RegressionModelType[]
+}
+
+const LinearRegressionViewAdaptor = (props: Props) => {
+
+  const model: PMML.RegressionModelType | undefined = getLinearRegressionModel(props);
```
Comment (kelvah) [inline front-end-poc/src/ModelLookup/pmml/LinearRegressionViewAdaptor.tsx:12]: I would not pass down the entire `props` object to a function if it is not needed, I think you could extract properties from `props` and then pass just what is consumed, `model` in this case. The code is also easier to understand this way. Do you agree?
The fact that `model` is declared here but it's different from the `props.model` is a bit confusing to me at first.

File: front-end-poc/src/Shared/api/audit.api.ts [hunk 3]
```diff
@@ -1,9 +1,10 @@
-import {callOnce, httpClient} from "./httpClient";
-import {AxiosRequestConfig} from "axios";
+import { callOnce, httpClient } from "./httpClient";
+import { AxiosRequestConfig } from "axios";
```
Comment (kelvah) [inline front-end-poc/src/Shared/api/audit.api.ts:2]: I really hope you did not fix all those spacing manually :D
We surely need a Prettier configuration with shared formatting rules. Will add it asap.

File: front-end-poc/src/ModelLookup/ModelDiagram.tsx [hunk 4]
```diff
@@ -1,22 +1,54 @@
...
+  executionId: string;
+}
+
+function makeDMNEditor(executionId: string): JSX.Element {
   const editorUrl = "https://kiegroup.github.io/kogito-online/?file=https://raw.githubusercontent.com/kiegroup/kogito-tooling/master/packages/online-editor/static/samples/sample.dmn#/editor/dmn";
   const kogitoIframe = () => {
-    return { __html: `<iframe src=${editorUrl} data-key="${selectedVersion.version}"></iframe>` };
+    return { __html: `<iframe src=${editorUrl} data-key="${executionId}"></iframe>` };
   };
+  return <><div className="model-diagram__iframe" dangerouslySetInnerHTML={kogitoIframe()} /></>;
```
[the remark this answers is not in the document]
Comment (manstis) [reply front-end-poc/src/ModelLookup/ModelDiagram.tsx:15]: Thanks. Fixed across all places I added single element fragments.

File: front-end-poc/src/ModelLookup/pmml/LinearRegressionView.tsx [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+        }
+        if (ticks[ticks.length - 1] !== end) {
+            ticks.push(end);
+        }
+        return ticks;
+    }
+
+    const modelName: string = props.modelName;
+    const width: number = props.width === undefined ? 500 : props.width;
+    const height: number = props.height === undefined ? 500 : props.height;
```
[the remark this answers is not in the document]
Comment (manstis) [reply front-end-poc/src/ModelLookup/pmml/LinearRegressionView.tsx:64]: Cool.. it's these types of trick I like to be told. 

I've changed it to the following to include `modelName` too:

`const { modelName = "undefined", width = 500, height = 500 } = props;`

File: front-end-poc/src/ModelLookup/pmml/LinearRegressionView.tsx [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+    height?: number
+    lines: Line[],
+    rangeX: Range,
+    rangeY: Range
+}
+
+const LinearRegressionView = (props: Props) => {
+
+    function roundedToFixed(_float: number, _digits: number): string {
+        var rounded = Math.pow(10, _digits);
```
[the remark this answers is not in the document]
Comment (manstis) [reply front-end-poc/src/ModelLookup/pmml/LinearRegressionView.tsx:42]: Sure thing.

File: front-end-poc/src/Shared/api/audit.api.ts (hunk 3 above)
Comment (manstis) [reply front-end-poc/src/Shared/api/audit.api.ts:2]: Nope, VSCode's default prettifier was responsible.

File: front-end-poc/src/ModelLookup/pmml/LinearRegressionViewAdaptor.tsx (hunk 2 above)
Comment (manstis) [reply front-end-poc/src/ModelLookup/pmml/LinearRegressionViewAdaptor.tsx:12]: I agree and have changed accordingly. 

I've also renamed the property to `models` too as it is an array.

File: front-end-poc/src/ModelLookup/ModelDiagram.tsx [hunk 7]
```diff
@@ -1,22 +1,54 @@
...
+}
+
+const ModelDiagram = (props: Props) => {
+
+  const { executionId } = props;
+  const [viewer, setViewer] = useState(<><Spinner size="xl" /></>);
+
+  useEffect(() => {
+    let didMount = true;
+    getModelDetail(executionId)
```
[the remark this answers is not in the document]
Comment (manstis) [reply front-end-poc/src/ModelLookup/ModelDiagram.tsx:29]: OK, I changed `ModelLookup` to do the retrieval and hold state with `ModelDiagram` simply rendering the correct viewer. I did try having state in both the `container` and `presentational` however it led to a infinite render loop. IDK why. Should I be looking at holding state in both components and fix the loop issue or is what I've now done OK?

**Comment by manstis** [comment]
@kelvah Fixes made in new commit. Thank-you.

File: front-end-poc/src/ModelLookup/ModelDiagram.tsx (hunk 7 above)
[the remark this answers is not in the document]
Comment (kelvah) [reply front-end-poc/src/ModelLookup/ModelDiagram.tsx]: No, it's ok. State should be lifted up to the highest level, so `ModelLookup`, no reason to handle it in both sides.

**Comment by kelvah** [comment]
@manstis Should we merge?

**Pull request merged**